src/lint/linter/UberCheckstyleLinter.php

```diff
@@ -102,37 +129,81 @@ private function checkScriptConfiguration() {
     }
   }
 
-  private function checkJarConfiguration() {
+  private function checkJavaConfiguration() {
     if (!Filesystem::binaryExists("java")) {
       throw new ArcanistMissingLinterException(
         pht('Java is not installed', get_class($this)));
     }
+  }
 
-    $absJarPath = Filesystem::resolvePath($this->checkstyleJar, $this->getProjectRoot());
-    if (!Filesystem::pathExists($absJarPath)) {
+  private function downloadCheckstyleJar($checkstyleJar, $checkstyleURL) {
+    $absJarPath = Filesystem::resolvePath($checkstyleJar, $this->getProjectRoot());
+    if(!is_writable(dirname($absJarPath))){
+      throw new InvalidArgumentException(
+        pht('Checkstyle jar cannot be written to "%s"', $checkstyleJar)
+      );
+    }
+
+    // Stage and validate to avoid file_put_contents leaving around an empty file
+    // if the URL is bad etc
+    $tmpFile = tempnam(sys_get_temp_dir(), 'TMP_');
+    $bytesDownloaded = file_put_contents($tmpFile, fopen($this->checkstyleURL, 'r'));
+    if ($bytesDownloaded === false || $bytesDownloaded == 0) {
```

Comment:
NIT:
`$bytesDownloaded == 0` is also true if `$bytesDownloaded == false` as:
```lang=php
>>> var_dump(0 == false);
bool(true)
```

It should be enough to check
```
if ($bytesDownloaded == 0) {

}
```